### ISSUE TITLE
[Redesign][Bug] Artists View shows all Artists in Offline Mode and Album Artists in Online Mode

### ISSUE THREAD

**Issue by lukaslindnermusic** [jmshrv/finamp#723]
Hi there,

this is still a thing, so I thought it might be good to create an issue for it, just for a better overview.

In Online/Default Mode, the „Artists“ tab shows the Album Artists (Artists associated with an entire album).
However, in Offline-Mode the „Artists“ tab shows all the artists (this includes features on individual tracks on an album or stuff like the orchestra on soundtracks that normally would just be considered to be the Artist but not the Album Artist (that would be the composer).).

I personally definitely prefer to browse album artists over artists, but I know that a few people like it the other way around.
I know that @Chaphasilor does not plan to create two separate tabs for Artists and Album Artists (do i remember correctly?), but what about making the Artists tab a) consistant in both modes (Online and Offline) and b) add a little toggle/button on the top right to either show all artists or just the album artists?
This setting should be remembered. 
If not possible or suitable in the top right section, maybe we could add it to the Settings?

Excited about your opinions on this 😊 
Thank you and have a nice day 😊

**Comment by Chaphasilor** [comment]
We're not constrained by the current design, the library view is going to get redesigned. So almost all options are on the table, including introducing a separate tab for artists (I'm more of an artists person myself, e.g. for listening to track featuring specific vocalists). But maybe there is an even better solution?

I'd like to try thinking outside the box. Can we somehow bring both types together? Would some kind of filter make sense? Do both types have the same properties? How are other apps handling this, both good and bad?

And yes, I'm considering the inconsistency between online and offline mode to be a bug, just haven't gotten around to looking into it.

**Comment by mihawk90** [comment]
> How are other apps handling this, both good and bad?

Not specifically for JF, but generally speaking most Android music apps don't handle Album Artists at all because they rely on Android's media framework which doesn't seem to do this.
So the only Apps that even have this are ones that handle the database side themselves (which in case of finamp of course is largely taken care of by JF).

Before I jumped to JF and Finamp I tried countless open source Apps that all had the aforementioned issue until after way too long I stumbled onto Musicolet (although not open source unfortunately), which did a lot of things right and ticked all my boxes.
Anyway, long story short Musicolet simply has an artists tab that has a secondary selection at the top as initially proposed here:
![Screenshot_20240616-185334](https://github.com/jmshrv/finamp/assets/556162/5b734750-4ce8-4458-b0ef-266084210180)

Other examples are foobar2000 for Android (although I didn't use it for long because I didn't like the interface), or Rocketplayer (with ads, yay).

**Comment by lukaslindnermusic** [comment]
Hey @Chaphasilor 😊 If nobody else wants do work on this, I would like to give it a shot. (#hackathon)
unfortunately I am busy this weekend, but I can work on it during next week.

I think this one goes hand in hand with https://github.com/jmshrv/finamp/issues/162.

It is marked as difficulty "moderate", so I am not sure if I am skilled enough, but as this is a thing that annoys me personally for a long time now, I am down to working through this, learning along the way and trying to find a solution.


In https://github.com/jmshrv/finamp/issues/723, there are a few different mockups as well.
I think, it basically comes down to deciding first if:

1. Having separate tabs/lists for "Album Artists" and "Artists"
2. Having one artists-tab/list/view and some sort of filtering where you can toggle between all artists and just album artists
    2a. Having a dropdown on the tab-button when long-pressing
    2b. Having a dedicated icon next to the sort-direction icon in the top right
    2c. Having a dedicated option "Show Album-Artists only" in the dropdown menu for the Sorting-by... maybe somehow separated by a horizontal line as it is a filter and not a sorting option. Can have a checkmark
    2d. Have a toggle in the settings (this would be bad if you want to switch in between both)

What would be the preferred solution? 
How hard would it be to implement? What would be the steps? 
Would it be too hard for a beginner finamp dev (but with programming background in other languages)?

**Comment by Chaphasilor** [comment]
Hey @lukaslindnermusic, feel free to go ahead! Even though I marked this as "moderate", I think it's absolutely doable for you. It's just not the best issue for someone just starting to contribute to Finamp, and there are a few design hurdles to overcome, as you are aware.

I'd suggest you either start with two separate tabs or a toggle button (could be one of the filter chips propsed by Erik), and then once the functionality is there, we can talk about how best to reflect it in the UI.  
We haven't settled for a concrete layout of the library for the redesign yet, so we're flexible there :)

**Comment by ingoratsdorf** [comment]
Not a Finamp contributor (yet) but definitely a music lover. Finamp was one of the reasons I switched from Plex and Emby back to JellyFin.
I would love to search for Album Artists because that's what I normally do. Maybe I don't have brains to remember individual tracks and their artists, but I do remember covers, thus the album artist.
Personally, I would not mind whether it would be 2 tabs or just a toggle, for as long as the latter one can be remembered. As Jellyfin features a separate tab, I'd like to think mimicking that would maybe best.

**Comment by lukaslindnermusic** [comment]
> Personally, I would not mind whether it would be 2 tabs or just a toggle, for as long as the latter one can be remembered. As Jellyfin features a separate tab, I'd like to think mimicking that would maybe best.

@ingoratsdorf yes, I thought about that, too. While it is a good idea most of the time to keep consistent with Jellyfin Web, I think, in this case, a toggle, or two buttons at the top make more sense... as both types are "Artists", just with different criteria. Album Artists and "Performing Artists" are closer together than Albums and Genres, for example. I think, it might be cleaner with a selection at the top of the artists list, where you can decide between "Album Artists" (which should be the default, I think) and "Performing Artists"... and yes, of course this setting should be remembered until you change it again.
And it should be consistent in both online and offline mode.

I'm now starting to work on it 😊 
Hope I can manage the task 😅

**Comment by Chaphasilor** [comment]
Fixed in [beta update 0.9.16](https://github.com/jmshrv/finamp/releases/tag/0.9.16-beta)!
